lib/obs_github_deployments/deployment.rb

```diff
@@ -19,6 +19,26 @@ def status
       # TODO: handle the possible exceptions properly
     end
 
+    def fail(reason:)
+      deployment = latest
+
+      if deployment.blank?
+        create_and_set_state(state: "failure", reason: reason)
+      else
+        add_state(deployment: latest, state: "failure", description: reason)
+      end
+    end
+
+    def succeed(reason:)
+      deployment = latest
+
+      if deployment.blank?
+        create_and_set_state(state: "success", reason: reason)
+      else
+        add_state(deployment: latest, state: "success", description: reason)
```

Comment:
@vpereira But know you will always change the state of the previous deployment, no matter what. Shouldnt you create a new one? Maybe only changing the state of the previous one if it is `inactive`